frontend/src/api/k8s/__tests__/notebooks.spec.ts

```diff
@@ -955,3 +956,30 @@ describe('removeNotebookSecret', () => {
     });
   });
 });
+
+describe('restartNotebook', () => {
+  it('should add restart notebook annotation', async () => {
+    const name = 'test-notebook';
+    const namespace = 'test-project';
+    const notebookMock = mockNotebookK8sResource({ uid });
+
+    k8sGetResourceMock.mockResolvedValue(notebookMock);
+    k8sPatchResourceMock.mockResolvedValue(notebookMock);
+
+    const renderResult = await restartNotebook(name, namespace);
+    expect(k8sPatchResourceMock).toHaveBeenCalledWith({
```

Comment:
Just a point of curiosity; in this test, why are you checking the arguments of the patch call and not asserting on the resulting CR after it was patched?

Essentially, I'm asking about this
> One can write unit tests either the classicist or the mockist way as per http://martinfowler.com/articles/mocksArentStubs.html
(quote from https://stackoverflow.com/questions/8666495/is-there-any-advantage-to-writing-both-classist-and-mockist-unit-tests-for-a-sin)

---

edit: never mind, I get it now; the patch is applied on the server (k8s api server), not in your program.